### DISCUSSION TITLE
Query regarding attaching objects to the robot and visualizing the usd

### DISCUSSION THREAD

**Discussion by shannonwarren** [NVlabs/curobo#234] [Software Q&A] [answered]
I'm following the simple_stacking example to attach objects to my ur5e. I create an `attached_object = Cuboid(name='test_box', pose=[0.2,0.2,0.7,1,0,0,0], dims=[0.05,0.05,0.05])` ; update my world and then use `self.motion_gen.attach_objects_to_robot` to attach the test_box. 

The problem I'm having it that the attached object is static (not deleted from my environment) in my saved usd and I expect it to move with the arm. Also, is there a way i could give the attached object a pose relative to the end effector link instead of the world frame.

Thank you,

**Comment by balakumar-s** [comment, marked as answer]
The usd viewer does not move the attached object. It rather only moves the spheres that represent the attached object. If you pass `visualize_robot_spheres=True` then you will see spheres representing the attached object.

To provide object pose relative to end-effector link, there is no direct API. You need to compute the transform to the robot's base from using kinematics. Let us know if you need help on this and we can share a small code snippet.
